### PR TITLE
Auto-Hide transactions from YNAB tracking accounts

### PR DESCRIPTION
**Pull request by tlmaloney**
Implement automatic hiding of transactions from YNAB tracking accounts
(off-budget accounts like 401k, mortgages, investment accounts). This
provides cleaner reports by default while still allowing users to toggle
visibility if needed.

Implementation:
- Add account cache to YNABClient to store account metadata
- Fetch and cache accounts during login (single API call)
- Check on_budget status when converting transactions
- Hide transactions from tracking accounts (on_budget=False)
- Clear account cache on close/logout

Testing:
- Add 9 comprehensive unit tests covering:
  * Account caching on login
  * Tracking account transactions hidden
  * Budget account transactions visible
  * Edge cases (missing cache, unknown accounts)
  * Combined hide conditions (deleted + transfer + tracking)
- All 1032 tests passing
- Type checking passes with 0 errors

### CODE REVIEW

File: docs/guide/ynab.md [hunk 1]
```diff
@@ -115,6 +115,49 @@ All edits sync back to YNAB immediately after commit. See the [Editing Guide](ed
...
+Since tracking account transactions don't affect your budget, they're filtered out by default to keep your reports
+focused on actual spending and income.
+
+### Viewing Hidden Transactions
+
+To see transactions from tracking accounts:
+
+1. Press **`h`** to toggle visibility of hidden transactions
+2. Hidden transactions will appear with a dimmed style
+3. Press **`h`** again to hide them
```
[the remark this answers is not in the document]
Comment (tlmaloney) [reply docs/guide/ynab.md]: Ah, YNAB doesn't give the user the option to hide transactions. You can hide categories from the current budget, I do that for some I no longer use. I didn't realize hidden transactions were coming from Monarch. For better or worse, I have these "non-budget" tracking accounts whose activity I wanted to auto-hide. Happy to let this one simmer a bit longer.

**Review by wesm — APPROVED** [review]
